compliance/collection/nodeinventorizer/caching_inventorizer_test.go

```diff
@@ -0,0 +1,128 @@
+package nodeinventorizer
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stretchr/testify/suite"
+)
+
+type mockSleeper struct {
+	receivedDuration time.Duration
+	callCount        int
+}
+
+func (ms *mockSleeper) mockWaitCallback(d time.Duration) {
+	ms.receivedDuration = d
+	ms.callCount++
+}
+
+type mockScanner struct{}
+
+func (m mockScanner) Scan(nodeName string) (*storage.NodeInventory, error) {
+	return &storage.NodeInventory{NodeName: nodeName}, nil
+}
+
+type TestComplianceCachingSuite struct {
+	suite.Suite
+}
+
+func TestComplianceCaching(t *testing.T) {
+	suite.Run(t, new(TestComplianceCachingSuite))
+}
+
+// run before each test
+// func (s *TestComplianceCachingSuite) SetupTest() {
+//	s.sleeper = mockSleeper{callCount: 0}
+//	s.cs = *NewCachingScanner(
+//		fmt.Sprintf("%s/inventory-cache", s.T().TempDir()),
+//		s.sleeper.mockWaitCallback,
+//	)
+//}
+
+func (s *TestComplianceCachingSuite) TestMin() {
+	initial := 2 * time.Second
+	maxBackoff := 10 * time.Second
+
+	actual := min(initial, maxBackoff)
+
+	// On successful test, the duration must not be overwritten
+	s.Equal(initial, actual)
+}
+
+func (s *TestComplianceCachingSuite) TestMinMaxBackoff() {
+	initial, _ := time.ParseDuration("100h")
+	maxBackoff, _ := time.ParseDuration("10s")
+
+	actual := min(initial, maxBackoff)
+
+	s.Equal(maxBackoff, actual)
+}
+
+func (s *TestComplianceCachingSuite) TestCalcNextBackoff() {
+	initial, _ := time.ParseDuration("2s")
+	cache := initial
+	maxBackoff, _ := time.ParseDuration("10s")
+	cs := *NewCachingScanner(mockScanner{}, "", cache, initial, maxBackoff, func(time.Duration) {})
+	expectedBackoff := initial * backoffMultiplier
```

Comment:
It isn't a great way to check multiplication operation results by doing multiplication and using same values.
Maybe here you just hardcode the expected result?

```suggestion
	expectedBackoff := (2 * time.Second) * 2
```